Cargo.toml

```diff
@@ -56,7 +56,7 @@ number_prefix = "0.4"
 openssl = { version = "0.10.48", optional = true }
 rand = "0.8.4"
 regex = "1.7.3"
-reqwest = { version = "0.11", features = ["json", "blocking", "stream"], optional = true }
+reqwest = { version = "0.11", features = ["json", "blocking", "stream", "rustls-tls", "trust-dns"], optional = true }
```

Comment:
We can do this by enable `opendal`'s `trust-dns` feature so that we don't need to play with `trust-dns-resolver` directly.

Reply:
`opendal`'s `trust-dns` feature simply enables `reqwest/trust-dns`.
There are currently no other ways to do it.

The reason for adding `trust-dns-resolver` as a dep to sccache is to enable extra features:
 - dnssec-ring to enable dnssec using ring
 - dns-over-rustls to enable dns-over-tls using ring
 - dns-over-https-rustls for dns over https
 - dns-over-quic for dns over http3 (quic)

The alternative to `ring` and `rustls` is `openssl`, and given that these extra features might come in handy for `sccache`, I think enabling them by default makes sense.